source/grid/tria.cc

```diff
@@ -11207,6 +11207,27 @@ Triangulation<dim, spacedim>::get_communicator() const
 
 
 
+template <int dim, int spacedim>
+const std::weak_ptr<const Utilities::MPI::Partitioner>
+Triangulation<dim, spacedim>::global_active_cell_index_partitioner() const
+{
+  return std::make_shared<const Utilities::MPI::Partitioner>(n_active_cells());
```

Comment:
I guess we never use this inside loops and rather only in basic circumstances? While the setup of an `MPI_COMM_SELF` partitioner should be cheap, it is not that cheap that we would like to do it millions of times.

Reply:
I don't see how this can work - won't the object always get deleted immediately since `weak_ptr` won't keep it alive? For example, if I do

```cpp
#include <memory>
#include <iostream>

std::weak_ptr<int>
get_42()
{
    return std::make_shared<int>(42);
}

int main()
{
    auto v = get_42();
    std::cout << "expired() = " << v.expired() << std::endl;
}
```

then the output is `expired() == 1`.